js/ViewportResizer.js

```diff
@@ -0,0 +1,223 @@
+  }
+
+  init () {
+    let self = this
+    // bind actions for viewport controls
+    // bind width input field
+    // init the viewport on load
+    let $viewportResizer = this.$ctx.nextElementSibling.querySelector(`.${this.selectors.iframe}`)
+    let $widthInput = this.$ctx.querySelector(`.${this.selectors.controls}-width`)
+    let $heightInput = this.$ctx.querySelector(`.${this.selectors.controls}-height`)
+
+    self.updateViewportContainer($viewportResizer)
+    // also update the input field with width
+    self.updateInputs($widthInput, $heightInput)
+
+    $widthInput.addEventListener('focusout', function () {
+      self.updateViewport({width: this.value})
+    })
+    // bind height input field
+    $heightInput.addEventListener('focusout', function () {
+      self.updateViewport({height: this.value})
+    })
+    // bind function to the reset button
+    this.$ctx.querySelector(`.${this.selectors.controls}-reset`).addEventListener('click', () => {
+      this.updateViewport({width: '', height: ''})
+    })
+    // bind function based on preset viewports
+    this.$ctx.querySelectorAll('[data-viewportsize]').forEach(function (element) {
+      element.addEventListener('click', function () {
+        self.updateViewport(JSON.parse(this.getAttribute('data-viewportsize')))
+      })
+    })
+    // bind to Styleguide:updateViewport eventlistener
+    this.$ctx.addEventListener('Styleguide:updateViewport', function (event) {
+      self.updateViewportContainer($viewportResizer)
+      // also update the input field with width
+      self.updateInputs($widthInput, $heightInput)
+    })
+    // listens for tab change event and removes or adds width of viewport
+    this.$ctx.nextElementSibling.querySelector(`.${this.selectors.iframe}`).addEventListener('Styleguide:tabChange', function (event) {
+      if (window.location.hash === '#example') {
+        this.style.width = `${self.viewportWidth}px`
+        this.style.height = `${self.viewportHeight}px`
+      } else {
+        this.parentNode.removeAttribute('style')
+      }
+    })
+
+    // bind viewport frame actions
+    this.$ctx.nextElementSibling._.contents($.create('div', {
+      className: `${this.selectors.example}-handle`
+    }))
+
+    this.$ctx.nextElementSibling.addEventListener('mousedown', function (event) {
+      self.initDrag(event)
+      this.setAttribute('data-resizable', '')
+    }, false)
+  }
+
+  initDrag (event) {
+    this.startX = event.clientX
+    this.startY = event.clientY
+    let $resizer = this.$ctx.nextElementSibling.querySelector(`.${this.selectors.iframe}`)
+
+    let computedStyle = document.defaultView.getComputedStyle($resizer)
+    this.startWidth = parseInt(computedStyle.width, 10)
+    this.startHeight = parseInt(computedStyle.height, 10)
+
+    let doc = document.documentElement
+    doc.addEventListener('mousemove', (event) => {
+      this.dragContainer(event, $resizer)
+    }, false)
+
+    doc.addEventListener('mouseup', () => {
```

Comment:
The width/height values of the steppers should update while dragging instead of after you release.